File: sender/src/domain/post.py
class Post:
    def __init__(self, dictionary):
        self.text = dictionary['text']
        self.preview_enabled = dictionary['previewEnabled']
        self.mode = self.get_parse_mode(dictionary['mode'])

    def get_parse_mode(self, mode):
        if mode == 'HTML':
            return 'HTML'
        elif mode == 'MARKDOWN':
            return 'Markdown'
        elif mode == "AS_IS":
            return "AS_IS"
        else:
            return None

    def __str__(self):
        return str(self.__dict__)


class PostGroup:
    def __init__(self, dictionary):
        self.channel_id = dictionary['channelId']
        self.channel_url = dictionary['channelUrl']
        self.post_id = dictionary['postId']
        self.posts = {t: Post(post) for t, post in dictionary['posts'].items()}

    def __str__(self):
        return str(self.__dict__)

File: tg/src/main/kotlin/com/github/telegram_bots/channels_feed/tg/domain/Post.kt
package com.github.telegram_bots.channels_feed.tg.domain

import com.github.badoualy.telegram.tl.api.TLMessage

data class ProcessedPost(
        val text: String? = null,
        val previewEnabled: Boolean = false,
        val mode: Mode
) {
    enum class Mode { AS_IS, TEXT, HTML, MARKDOWN }
}

data class ProcessedPostGroup(
        val channelId: Int,
        val channelUrl: String,
        val postId: Int,
        val posts: Map<Type, ProcessedPost>
) {
    enum class Type { FULL, SHORT, TITLE_ONLY }
}

data class RawPostData(val raw: TLMessage, val channel: Channel)
